### PR TITLE
DataFixtures - part1 - translations moved into separated files

### PR DESCRIPTION
**Pull request by Miroslav-Stopka**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Installation with single cs domain fails as it is not possible to find countries by names "Czech republic"/"Slovakia" in `UserDataFixture` (In a fact, there are countries "Česká republika"/"Slovensko" for the Czech locale). Another problem is a very complicated extension of data fixtures with a new locale - as a developer needs to modify all the files with data fixtures, where the multilang attributes are set (eg. see https://github.com/shopsys/shopsys/blob/7.2/project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/CategoryDataFixture.php#L69)
|New feature| No
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No
|Fixes issues| ... 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by vitek-rostislav** [comment]
Hi @Miroslav-Stopka, thanks for the PR! :heart:  I will make sure it is processed during the next sprint that starts the next Thursday :wink:

**Comment by vitek-rostislav** [comment]
Hi @Miroslav-Stopka, after today's sprint planning, I have news for you - this PR is not planned into the current sprint due to other priorities, however, it is on the top of our buffer so it will be processed if there is some time left :wink:

File: project-base/src/Shopsys/ShopBundle/DataFixtures/resources/demo-data-customers.csv [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-Alexandr;Ton;no-reply.9@shopsys.com;no-reply.9;false;;;;Na Strzi 3;Bohum�n;69084;;Czech republic;false;;;;;;;;;1
-Pavel;Nedv�d;no-reply.10@shopsys.com;no-reply.10;false;;;;Tur�nsk� 5;Tur�n;12345;606060606;Czech republic;true;Bahamy;;;;99999;Bahamsk� 99;;Czech republic;1
-Rostislav;V�tek;vitek@shopsys.com;user123;true;Shopsys;123456;123456;Hlubinsk�;Ostrava;70200;605000123;Czech republic;true;Ostrava;Rockpoint;Eva;Wallicov�;70030;Rudn�;123456789;Czech republic;1
-Johny;English;no-reply@shopsys.com;user123;true;Shopsys;123456;;Hlubinsk�;Ostrava;70200;603123456;�esk� republika;false;London;;;;;;;;2
-�ubom�r;Nov�k;No-reply.11@shopsys.com;test123;false;;;;Brn�nsk�;Bratislava;1010;;Slovakia;true;Bratislava;;;;10100;Ostravsk� 55/65A;758686320;Slovakia;1
+firstName;lastName;email;password;company;companyName;companyNumber;companyTaxNumber;street;city;postcode;telephone;countryCode;deliveryAddressFilled;dCity;dCompanyName;dFirstName;dLastName;dPostcode;dStreet;dTelephone;dCountryCode;domainId
+Jarom�r;J�gr;no-reply@shopsys.com;user123;true;Shopsys;123456;;Hlubinsk�;Ostrava;70200;605000123;CZ;false;;;;;;;;;1
```
Comment (vitek-rostislav) [inline project-base/src/Shopsys/ShopBundle/DataFixtures/resources/demo-data-customers.csv]: seems like there is something wrong with the encoding :thinking:

**Comment by henzigo** [comment]
Hello, what about this PR? I'm sad that this not your priority. It's already takes over 2 months and it would make a lot of work easier for implementators on SSFW.

File: project-base/src/Shopsys/ShopBundle/DataFixtures/resources/demo-data-customers.csv (hunk 1 above)
Comment (pk16011990) [reply project-base/src/Shopsys/ShopBundle/DataFixtures/resources/demo-data-customers.csv]: This file is in CP1250 for easier work in MS Excel. Work with another encoding in MS Excel is uncomfortable... And Libre office/Open Office can open any encoding easily.

**Comment by vitek-rostislav** [comment]
Hi @henzigo, sadly, it has been in our buffer for quite a long time, however, there are still more important things to do..I am trying to push it

**Comment by Miroslav-Stopka** [comment]
Hello, what about this PR? Is this PR dead already :) ?

**Comment by vitek-rostislav** [comment]
Hello Mr. @Miroslav-Stopka (my greetings to Mr. @miroslav2stopka as well :tophat:)

Please accept my apologies for the late response...

This PR is not dead, however, we are kind of struggling with handling the whole topic (preparation of tests and data fixtures for various domain&locale settings). 

We have described it in a separate user-story in our internal backlog (US-6365), and it has been evaluated to 20 story-points. I am still trying to push the user-story :wink: Once the user-story is finished, this PR will be closed some way (merged, reworked, used as an inspiration, rejected with an explanation and link to a new solution...)

**Comment by Miroslav-Stopka** [comment]
Hi @vitek-rostislav,

thank you for your response. Please, just dont forget on the main usage scenarios and on the primary target group.

I mean: 

1) Modifications/commits/merges already created data vs translate all data into new language? Which one will be the most occuring scenario? Which one will be occuring only once per year?

2) Developers vs project managers? Which one will be in touch with this funcionality much offen? Who will modify this data much offen?

Please consider this, the first step should be the evaluation of scenarios, the second should be the evaluation of target groups. And only after this steps (not before them), there can come the third step (so this step should not be the first one) - the possible solutions for the findings and problems from first two steps.  :)

Good Luck

**Comment by vitek-rostislav** [comment]
Hi @Miroslav-Stopka, just FYI, this PR should be closed soon by https://github.com/shopsys/shopsys/pull/1425. Thanks for pushing this topic! We decided not to use your solution as is because this approach is quite complicated, you are basically implementing your own translator here. But we used your idea and leveraged "natural" translator in data fixtures.

**Pull request closed without merging**